TYPO3.Flow/Classes/TYPO3/Flow/Utility/Environment.php

```diff
@@ -58,28 +47,22 @@ public function __construct(\TYPO3\Flow\Core\ApplicationContext $context)
      *
      * @param string $temporaryDirectoryBase Base path of the temporary directory, with trailing slash
      * @return void
+     * @throws Exception
+     * @deprecated since 2.3 - Set the environment variable FLOW_PATH_TEMPORARY_BASE to change the temporary directory base, see Bootstrap::defineConstants()
      */
     public function setTemporaryDirectoryBase($temporaryDirectoryBase)
     {
-        $this->temporaryDirectoryBase = $temporaryDirectoryBase;
-        $this->temporaryDirectory = null;
+        throw new Exception('Changing the temporary directory path during runtime is no longer supported. Set the environment variable FLOW_PATH_TEMPORARY_BASE to change the temporary directory base', 1441355116);
     }
 
     /**
      * Returns the full path to Flow's temporary directory.
      *
-     * @return string Path to PHP's temporary directory
-     * @api
```

Comment:
I didn't deprecated it because it is used in the core quite a lot - and replacing it with the constant is not a (viable) option because many tests that mock the environment would fail..
I removed the api annotation because it was the last method with it and I think that we should get rid of the whole Environment class at some point.. But, sure, we could deprecated it in a follow-up targeted for 3.1+ - agree?